streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/testing/TestRunSourceSpout.java

```diff
@@ -104,13 +100,12 @@ public void nextTuple() {
         }
 
         if (emitCount == 0) {
-            LOG.info("No more records, sleeping...");
+            LOG.info("All iterations are completed or in sleep phase, sleeping...");
             try {
                 Thread.sleep(1000);
```

Comment:
Say if the user gives 500 millis sleep time, is theres a chance that it will not be honored since the sleep here is hard-coded?

Reply:
@arunmahadevan 
Yes that's right and I forgot it. But given that we need to care about multiple streams, I would handle this as making sleep more tight (say 50~100 ms). What do you think?

Reply:
We could also calculate common divisor and apply that as well, but just make it sleeping less is viable. Given that sleeping also doesn't ensure waking up after exact period, sleeping slightly more shouldn't hurt.